torch_xla/experimental/_distributed_checkpoint_helpers.py

```diff
@@ -186,4 +191,36 @@ def narrow_tensor_by_index(tensor: torch.Tensor, offsets: Sequence[int],
       # recording here for the narrow op and 'local_shard' should be a
       # leaf variable in the autograd graph.
       narrowed_tensor = narrowed_tensor.narrow(idx, offset, size)
-  return narrowed_tensor
\ No newline at end of file
+  return narrowed_tensor
+
+
+def _is_sharded_tensor(x: Any) -> bool:
+  """Return true if the tensor's data is sharded across multiple devices"""
+  return isinstance(
+      x, XLAShardedTensor) and x.sharding_type != ShardingType.REPLICATED
+
+
+def _unwrap_xla_sharded_tensor(x: Any) -> Any:
+  if isinstance(x, XLAShardedTensor):
+    return x.global_tensor
+  return x
+
+
+@dataclasses.dataclass
+class _CpuShards:
+  shards: List[xs.XLAShard]
+  global_shape: torch.Size
+
+
+def _sharded_cpu_state_dict(state_dict: STATE_DICT_TYPE) -> STATE_DICT_TYPE:
+  """
+  Converts a state_dict on XLA device to a sharded state_dict on CPU.
+  """
+
+  def move_state_dict_to_cpu(v):
+    v = xs.wrap_if_sharded(v)
+    if not _is_sharded_tensor(v):
+      return _unwrap_xla_sharded_tensor(v).cpu()
```

Comment:
We should generalize this some - the state dict may contain non-Tensor fields like scalars, so we should only call `.cpu` if the result of `_unwrap` is a tensor.